crates/module-system/sov-state/src/cache.rs

```diff
@@ -522,22 +560,25 @@ impl<N: ProvableCompileTimeNamespace> ProvableStorageCache<N> {
     }
 
     /// Replaces the keyed value on the storage.
-    pub fn set(&mut self, key: &SlotKey, value: SlotValue, rollup_height: u64) {
-        self.cache
-            .add_write(key.clone(), Some(value), rollup_height);
+    pub fn set(&mut self, key: &SlotKey, value: SlotValue) {
+        self.cache.add_write(key.clone(), Some(value));
     }
 
     /// Deletes a keyed value from the cache.
-    pub fn delete(&mut self, key: &SlotKey, rollup_height: u64) {
-        self.cache.add_write(key.clone(), None, rollup_height);
+    pub fn delete(&mut self, key: &SlotKey) {
+        self.cache.add_write(key.clone(), None);
     }
 
     // This method can be called only once per given key.
-    fn add_read(&mut self, key: SlotKey, node: Option<NodeLeafAndMaybeValue>) {
-        self.revertable_ordered_reads
-            .push((key.clone(), node.as_ref().map(|n| n.leaf)));
-
-        self.cache.add_read(key, node);
+    fn add_read(
+        key: SlotKey,
```

Comment:
What is the reason this method (and others) no longer takes &mut self but instead accepts revertable_ordered_reads as a parameter?